wagtail/core/actions/create_alias.py

```diff
@@ -0,0 +1,253 @@
+import logging
+import uuid
+
+from django.core.exceptions import PermissionDenied
+
+from wagtail.core.log_actions import log
+from wagtail.core.models.copying import _copy, _copy_m2m_relations
+from wagtail.core.models.i18n import TranslatableMixin
+
+
+logger = logging.getLogger("wagtail.core")
+
+
+class CreatePageAliasIntegrityError(RuntimeError):
+    """
+    Raised when creating an alias of a page cannot be performed for data integrity reasons.
+    """
+
+    pass
+
+
+class CreatePageAliasPermissionError(PermissionDenied):
+    """
+    Raised when creating an alias of a page cannot be performed due to insufficient permissions.
+    """
+
+    pass
+
+
+class CreatePageAliasAction:
+    """
+    Creates an alias of the given page.
+
+    An alias is like a copy, but an alias remains in sync with the original page. They
+    are not directly editable and do not have revisions.
+
+    You can convert an alias into a regular page by setting the .alias_of attribute to None
+    and creating an initial revision.
+
+    :param recursive: create aliases of the page's subtree, defaults to False
+    :type recursive: boolean, optional
+    :param parent: The page to create the new alias under
+    :type parent: Page, optional
+    :param update_slug: The slug of the new alias page, defaults to the slug of the original page
+    :type update_slug: string, optional
+    :param update_locale: The locale of the new alias page, defaults to the locale of the original page
+    :type update_locale: Locale, optional
+    :param user: The user who is performing this action. This user would be assigned as the owner of the new page and appear in the audit log
+    :type user: User, optional
+    :param log_action: Override the log action with a custom one. or pass None to skip logging, defaults to 'wagtail.create_alias'
+    :type log_action: string or None, optional
+    :param reset_translation_key: Generate new translation_keys for the page and any translatable child objects, defaults to False
+    :type reset_translation_key: boolean, optional
+    """
+
+    def __init__(
+        self,
+        page,
+        *,
+        recursive=False,
+        parent=None,
+        update_slug=None,
+        update_locale=None,
+        user=None,
+        log_action="wagtail.create_alias",
+        reset_translation_key=True,
+        _mpnode_attrs=None
+    ):
+        self.page = page
+        self.recursive = recursive
+        self.parent = parent
+        self.update_slug = update_slug
+        self.update_locale = update_locale
```

Comment:
When creating a page that's not at the root, the locale should always match the parent.

By default, the locale would be changed to match the parent's locale when the page is saved: https://github.com/wagtail/wagtail/blob/main/wagtail/core/models/__init__.py#L439

This would mean that an invalid value in ``update_locale`` would be ignored. So I wonder if we should check for this? Perhaps we should prevent it from being used if the destination is not the root page?

```python
if not parent.is_root() and self.update_locale is not None:
    raise CreatePageAliasIntegrityError(
        "Cannot manually set locale of a page that's not at the root of the tree."
    )
```

Reply:
There might be a bug in the underlying implementation here. If you set ``update_locale`` to something incorrect, it will silently change the page's locale back to the valid value on save, but I'm not sure if it would do this for any child objects.

Reply:
> Perhaps we should prevent it from being used if the destination is not the root page?

I've added the check you proposed.

> I'm not sure if it would do this for any child objects.

I will try to investigate this.

Reply:
> I've added the check you proposed. However it makes many tests fail currently...

Oh, let's leave it how it is and investigate in a separate PR

Reply:
>Oh, let's leave it how it is and investigate in a separate PR

Alright!